src/Microsoft.DotNet.Build.Tasks/PackageFiles/tests.targets

```diff
@@ -145,24 +171,64 @@
       <TargetingPackExclusions Include="@(ReferenceFromRuntime)"/>
 
-      <!-- Whitelisted runtime assemblies that are OK to reference. -->
-      <ReferenceFromRuntime Include="xunit.core" />
-      <ReferenceFromRuntime Include="Xunit.NetCore.Extensions" />
-      <ReferenceFromRuntime Include="xunit.assert" />
-      <ReferenceFromRuntime Include="xunit.abstractions" />
-      <ReferenceFromRuntime Include="xunit.performance.core" />
-      <ReferenceFromRuntime Include="xunit.performance.api" />
-
       <!-- Reference everything in the targeting pack directory -->
       <TargetingPackItems Include="%(TargetingPackDirs.Identity)/*.dll" />
 
       <DefaultReference Include="%(TargetingPackItems.Filename)" Exclude="@(TargetingPackExclusions)" />
     </ItemGroup>
+
+    <!-- Default test references that are ok to reference. -->
+
+    <!-- xunit -->
+    <ItemGroup Condition="'$(IncludeTestReferences)' == 'true'">
+      <ReferenceFromRuntime Include="Microsoft.DotNet.XUnitExtensions" />
+      <ReferenceFromRuntime Include="xunit.abstractions" />
+      <ReferenceFromRuntime Include="xunit.assert" />
+      <ReferenceFromRuntime Include="xunit.core" />
+    </ItemGroup>
+
+    <!-- netcoreapp -->
+    <ItemGroup Condition="'$(IncludePerformanceTestReferences)' == 'true' AND '$(BuildingNETCoreAppVertical)' == 'true'">
+      <ReferenceFromRuntime Include="xunit.performance.api" />
+      <ReferenceFromRuntime Include="xunit.performance.core" />
+    </ItemGroup>
+
+    <ItemGroup Condition="'$(IncludeVSTestReferences)' == 'true' AND '$(BuildingNETCoreAppVertical)' == 'true'">
+      <ReferenceFromRuntime Include="Microsoft.DotNet.PlatformAbstractions" />
+      <ReferenceFromRuntime Include="Microsoft.Extensions.DependencyModel" />
+      <ReferenceFromRuntime Include="Microsoft.TestPlatform.CommunicationUtilities" />
+      <ReferenceFromRuntime Include="Microsoft.TestPlatform.CoreUtilities" />
+      <ReferenceFromRuntime Include="Microsoft.TestPlatform.CrossPlatEngine" />
+      <ReferenceFromRuntime Include="Microsoft.TestPlatform.PlatformAbstractions" />
+      <ReferenceFromRuntime Include="Microsoft.VisualStudio.CodeCoverage.Shim" />
+      <ReferenceFromRuntime Include="Microsoft.VisualStudio.TestPlatform.Common" />
+      <ReferenceFromRuntime Include="Microsoft.VisualStudio.TestPlatform.ObjectModel" />
+      <ReferenceFromRuntime Include="testhost" />
+      <ReferenceFromRuntime Include="xunit.runner.visualstudio.dotnetcore.testadapter" />
+    </ItemGroup>
+
+    <!-- uap -->
+    <ItemGroup Condition="'$(IncludeVSTestReferences)' == 'true' AND ('$(BuildingUAPVertical)' == 'true' OR '$(BuildingUAPAOTVertical)' == 'true')">
+      <ReferenceFromRuntime Include="testhost" />
+      <ReferenceFromRuntime Include="xunit.runner.visualstudio.uwp.testadapter" />
+    </ItemGroup>
+
+    <!-- netfx -->
+    <ItemGroup Condition="'$(IncludePerformanceTestReferences)' == 'true' AND '$(BuildingNETFxVertical)' == 'true'">
```

Comment:
This can be unified with the netcoreapp one. I understand your intention of having them separate -- but that brings duplication.